.github/workflows/cypress.yaml

```diff
@@ -170,3 +187,66 @@ jobs:
+    runs-on: ubuntu-latest
+    if: ${{ needs.prepare.outputs.testrail_enable }}
+    steps:
+      - uses: actions/download-artifact@v3
+        with:
+          name: cypress-junit
+      - name: Prepare testrail upload config
+        id: testrailprep
+        env:
+          TESTRAIL_PROJECT_ID: ${{ secrets.TESTRAIL_PROJECT_ID }}
+          TEST_ID: ${{ github.run_id }}
+          TESTRAIL_URL: https://elementqa.testrail.io
+          TESTRAIL_USER: ${{ secrets.TESTRAIL_USER }}
+          TESTRAIL_API_KEY: ${{ secrets.TESTRAIL_API_KEY }}
+        run: |
+          echo '{"name": "element-web cypress '$TEST_ID'"}' > body.json # TODO add description with more context?
+          RUN_ID=`curl -X POST -d @body.json -u "$TESTRAIL_USER:$TESTRAIL_API_KEY" -H "Content-Type: application/json" "$TESTRAIL_URL/index.php?/api/v2/add_run/$TESTRAIL_PROJECT_ID" | jq '.id'`
+          PROJECT_NAME=`curl -X GET -u "$TESTRAIL_USER:$TESTRAIL_API_KEY" -H "Content-Type: application/json" "$TESTRAIL_URL/index.php?/api/v2/get_project/$TESTRAIL_PROJECT_ID" | jq '.name'`
+          echo "run_id=$RUN_ID" >> $GITHUB_OUTPUT
+          echo "project_name=$PROJECT_NAME" >> $GITHUB_OUTPUT
+      - name: setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - run: pip install trcli
+      - name: Upload junit files
+        env:
+          TESTRAIL_PROJECT_ID: ${{ secrets.TESTRAIL_PROJECT_ID }}
+          TESTRAIL_URL: https://elementqa.testrail.io
+          TESTRAIL_USER: ${{ secrets.TESTRAIL_USER }}
+          TESTRAIL_API_KEY: ${{ secrets.TESTRAIL_API_KEY }}
+          TESTRAIL_RUN_ID: ${{ steps.testrailprep.outputs.run_id }}
+        run: |
+          for file in results-*.xml; do
+            echo "Handling $file"
+            trcli -y -h $TESTRAIL_URL \
+                  --project-id $TESTRAIL_PROJECT_ID \
+                  --project ${{ steps.testrailprep.outputs.project_name }} \
+                  --username $TESTRAIL_USER \
+                  --password $TESTRAIL_API_KEY \
+                  parse_junit \
+                  --run-id $TESTRAIL_RUN_ID \
+                  --title "if you see this please check cypress build for run id not being provisioned" \
+                  -f $file
+          done
+      - name: Close test run
+        id: testrailpost
+        if: always()
+        env:
+          TESTRAIL_URL: https://elementqa.testrail.io
+          TESTRAIL_USER: ${{ secrets.TESTRAIL_USER }}
+          TESTRAIL_API_KEY: ${{ secrets.TESTRAIL_API_KEY }}
+          TESTRAIL_RUN_ID: ${{ steps.testrailprep.outputs.run_id }}
+        run: |
+          CLOSE_RESPONSE=`curl -X POST -d '{}' -u "$TESTRAIL_USER:$TESTRAIL_API_KEY" -H "Content-Type: application/json" "$TESTRAIL_URL/index.php?/api/v2/close_run/$TESTRAIL_RUN_ID"`
+          if [ ! "0" == "`echo $CLOSE_RESPONSE | jq .untested_count`" ] ; then echo "::warning title=Missing Tests::Testrail reported some cypress tests were not run. $CLOSE_RESPONSE"; fi
```

Comment:
Yeah, as i mentioned above, I'd want to do this properly as part of the second use of testrail rather than up front, we've never used testrail so far, and I'm not even sure if this is the best interface to be using the testrail API.

The right way to do the action is actually to reimplement a chunk of trcli without some of the design decisions they've made, then actually have a proper action with a post step that closes the run correctly, etc, but doing the action in a correctly reusable way is more than I want to do until we've seen it work for a while.

I suppose we could copy it into an action in `.github/actions/testrail-composite` or something, but that's just shuffling the same code around.

I am happy to commit to doing this when we get the next project's reports into testrail (so knowing what the reusable interface looks like). At that point the above would turn into an artifact download + invocation of a single action, and the action wouldn't have any of this bash / python in it.

Is this a blocker to merging?